src/jit/lsraarmarch.cpp

```diff
@@ -1052,6 +928,75 @@ void Lowering::TreeNodeInfoInitBlockStore(GenTreeBlk* blkNode)
     }
 }
 
+//------------------------------------------------------------------------
+// GetIndirSourceCount: Get the source registers for an indirection that might be contained.
+//
+// Arguments:
+//    node      - The node of interest
+//
+// Return Value:
+//    The number of source registers used by the *parent* of this node.
+//
+int Lowering::GetIndirSourceCount(GenTreeIndir* indirTree)
+{
+    int      srcCount = 0;
+    GenTree* addr     = indirTree->gtGetOp1();
+    GenTree* base     = addr;
+    GenTree* index    = nullptr;
+
+    if (addr->isContained())
```

Comment:
Sounds reasonable; I'll make this and the change below after measuring throughput.